### PR TITLE
Add batch file for easy build

### LINKED ISSUE #2274: [REQUEST] Add batch file for normal user(also for developer easier build)
**Is your feature request related to a problem? Please describe.**

**Describe the feature you'd like to see implemented**

add some batch file for build / download / update=

**References**

**Additional context**

Discord username (if different from GitHub):

### PR DESCRIPTION
**Pull request by Kimbyeoungjang**
Fixes #2274

## Summary of Changes

Add batch file for easy build, update, etc... for user

## Screenshots (if necessary)

image

## Additional context

Discord username (if different from GitHub):

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

**Comment by Kimbyeoungjang** [comment]
please ignore upper commit

File: installer.cmd [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+if "%contactChoose%"=="1" (
+start "" https://discord.com/invite/flybywire & goto main
+) else if "%contactChoose%"=="2" (
+start "" https://github.com/flybywiresim/a32nx & goto main
+) else if "%contactChoose%"=="3" (
+start "" https://opencollective.com/flybywire & goto main
+) else if "%contactChoose%"=="4" (
+goto main
+) else (
+goto conatct)
```
Comment (davwheat) [inline installer.cmd:250]:
```suggestion
goto contact)

```

File: installer.cmd [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+)
+
+goto main
+
+:dockerERROR
+cls
+echo.
+echo ===================================================================
+echo              You should install Docker for build
+echo                Do you wanna install Docker?
```
Comment (davwheat) [inline installer.cmd:45]: Should we be more formal here?

(i.e. `wanna` -> `want to`)

File: .github/CHANGELOG.md [hunk 3]
```diff
@@ -138,6 +138,7 @@
 1. [ECAM] Add icing-related ECAM warnings and memos - @beheh (Benedict Etzel)
 1. [CDU] Fix CLR position in scratchpad - @beheh (Benedict Etzel)
 1. [General] Added INIT BARO option, INHG/HPA/AUTO - @theomessin (Theodore Messinezis)
+1. [BAT] Add Batchfile for easy development and build - @Kimbyeongjang (김병장#7165)
```
Comment (davwheat) [inline .github/CHANGELOG.md:141]:
```suggestion
1. [BAT] Add batch file to automate download and building A32NX from source for non-experienced users - @Kimbyeongjang (김병장#7165)
```

File: installer.cmd [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+echo ===================================================================
+echo.
+echo
+set /p docker= Y or N : 
+if "%docker%"=="Y" start "" https://www.docker.com/get-started & goto exit
```
Comment (davwheat) [inline installer.cmd:50]: You can use this to check case-insensitively. That way you don't need to check against `y` and `Y`.

```batch
if /i "%docker%"=="y"
```

File: installer.cmd [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+if "%docker%"=="y" start "" https://www.docker.com/get-started & goto exit
+if "%docker%"=="N" goto exit
+if "%docker%"=="n" goto exit
```
Comment (davwheat) [inline installer.cmd:53]: What if the user inputs `x`? Won't the script just continue executing downwards? We should ask for input again.

(This applies to the other error sections too)

```suggestion
if "%docker%"=="n" (goto exit)
else goto dockerERROR
```

File: installer.cmd [hunk 6]
```diff
@@ -0,0 +1,253 @@
+::@Kimbyeongjang - (김병장#7165)
+title A32NX Builder
+@echo off
```
Comment (davwheat) [inline installer.cmd:3]: We should put `@echo off` before the title command, otherwise the command is printed to the cmd window.

**Review by davwheat — CHANGES_REQUESTED** [review]

**Comment by Kimbyeoungjang** [comment]
thx for review @davwheat. i fix all that review :)

File: installer.cmd [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+    goto UACPrompt
+) else ( goto gotAdmin )
+
+:UACPrompt
+    echo Set UAC = CreateObject^("Shell.Application"^) > "%temp%\getadmin.vbs"
+    set params = %*:"=""
+    echo UAC.ShellExecute "cmd.exe", "/c %~s0 %params%", "", "runas", 1 >> "%temp%\getadmin.vbs"
+    "%temp%\getadmin.vbs"
+    rem del "%temp%\getadmin.vbs"
+    exit /B
```
Comment (davwheat) [inline installer.cmd:17]:
```suggestion
    exit
```

No need for `/B` as we're [not setting an error code](https://ss64.com/nt/exit.html).

File: installer.cmd [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+cls
+echo.
+echo ===================================================================
+echo               You should install Docker for build
+echo                 Do you want to install Docker?
+echo ===================================================================
+echo.
+echo
+set /p docker= Y or N : 
+if "%docker%"=="y" (
```
Comment (davwheat) [inline installer.cmd:49]: You need the `/i` here to handle uppercase (e.g. `Y`) too. Learn more here: https://ss64.com/nt/if.html

This also applies further down.

```suggestion
if /i "%docker%"=="y" (
```

**Review by davwheat — CHANGES_REQUESTED** [review]

**Comment by Highflyer2010** [comment]
what is this actually for?

**Comment by Kimbyeoungjang** [comment]
> what is this actually for?

First time, i just build this batch file for non-experienced user about coding.
But there is already dev build installer, so this is going to trashbin i think XD

**Comment by Highflyer2010** [comment]
> 
> 
> > what is this actually for?
> 
> First time, i just build this batch file for non-experienced user about coding.
> But there is already dev build installer, so this is going to trashbin i think XD

ah now i get it. With us having the installer i don't think it's really needed now

**Comment by Kimbyeoungjang** [comment]
> > > what is this actually for?
> > 
> > 
> > First time, i just build this batch file for non-experienced user about coding.
> > But there is already dev build installer, so this is going to trashbin i think XD
> 
> ah now i get it. With us having the installer i don't think it's really needed now

Yes i think. Closed pr.

**Pull request closed without merging**